src/spec.c

```diff
@@ -1181,11 +1181,12 @@ static void IndexSpecCache_Free(IndexSpecCache *c) {
 // and at this point the refcount only gets decremented so there is no wory of some thread increasing the
 // refcount while we are freeing the cache.
 void IndexSpecCache_Decref(IndexSpecCache *c) {
-  if (!__atomic_sub_fetch(&c->refcount, 1, __ATOMIC_RELAXED)) {
+  if (c && !__atomic_sub_fetch(&c->refcount, 1, __ATOMIC_RELAXED)) {
```

Comment:
in which case it becomes NULL?

Reply:
All the places that can hold the cache today might not has one, so they all had a check if it’s not NULL before calling this. 
I removed the need to check if their possible cache is null or not